server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/component/BranchPersisterImpl.java

```diff
@@ -78,12 +79,25 @@ private static void checkState(boolean condition, String msg) {
   private static BranchDto toBranchDto(ComponentDto componentDto, Branch branch) {
     BranchDto dto = new BranchDto();
     dto.setUuid(componentDto.uuid());
+
     // MainBranchProjectUuid will be null if it's a main branch
     dto.setProjectUuid(firstNonNull(componentDto.getMainBranchProjectUuid(), componentDto.projectUuid()));
     dto.setKey(branch.getName());
     dto.setBranchType(branch.getType());
+
     // merge branch is only present if it's a short living branch
     dto.setMergeBranchUuid(branch.getMergeBranchUuid().orElse(null));
+
+    if (branch.getType() == BranchType.PULL_REQUEST) {
+      // TODO this entire object should probably come from Branch.getPullRequestData
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fjanos%2FSONAR-10365%2Fpopulate-pull-request-column&issues=AWGuwEC_aHWQ5zPscll0&open=AWGuwEC_aHWQ5zPscll0)

[ISSUE_KEY]: AWGuwEC_aHWQ5zPscll0